pwa-devdocs/src/technologies/basic-concepts/internationalization/index.md

```diff
@@ -0,0 +1,157 @@
+
+Example `en_US.json` content:
+
+```json
+{
+    "About Us": "About Us",
+    "Hi, {name}": "Hi, {name}",
+    "accountMenu.orderHistoryLink": "Order History",
+    "autocomplete.resultSummary": "{resultCount} items"
+}
+```
+
+### ID formats
+
+The JSON object's keys act as unique IDs for localized text.
+The example posted shows the different ways you can identify the localized text in a dictionary file.
+
+The first and second entry uses the original `en_US` locale text as the ID for the translated text.
+This is the same approach used in Magento.
+
+The third and fourth entry uses an alternative, dot notation approach for defining the ID for translated text.
+This is the main format used in PWA Studio components, and it helps identify the component and element where the text appears.
+
+Both approach have their pros and cons, and developers are free to choose which approach works for them when they develop their own components and storefronts.
+
+### Message syntax
+
+The i18n framework accepts the same [message syntax][] as the underlying `react-intl` library.
+Along with static text, this syntax supports variables, dates, and even conditional formatting.
+
+To translate text with variables, pass in a mapping object to the `values` prop in the `FormattedMessage` component.
+
+{% raw %}
+
+```jsx
+return(
+    <FormattedMessage
+        id="component.greeting"
+        defaultMessage="Hi, {name}"
+        values={{
+            name: 'Veronica',
+        }}
+    />
+)
+```
+
+{% endraw %}
+
+When using the `formatMessage()` function, pass in the mapping object as the second parameter.
+
+```jsx
+const text = formatMessage(
+    { id:"component.greeting", defaultMessage="Hi, {name}"},
+    { name: currentUser.firstname }
+);
+```
+
+For more details, see the [message syntax][] documentation at FormatJS.
+
+## Language packages and plugins
+
+Language packages provide translation data for one or more locales.
+They are also used to override the text in the same locale.
+
+Unlike the core Magento application, which install language packages through Composer, PWA Studio storefronts install language packages as NPM dependencies.
+
+An NPM dependency is a language package if it meets the following criteria:
+
+- The package contains an intercept file
+- The intercept file sets the special feature `i18n` flag to `true` for the package
+- The package contains an `i18n` directory
+- The `i18n` directory contains a dictionary file with a locale formatted name
+
+<!-- TODO: Create an in-depth tutorial for creating a language package extension -->
+
+## Build process
+
+To optimize runtime performance, the i18n framework compiles all the translation data during the build process.
+
+The following is a high level summary of the actions the i18n framework takes to compile the translation data:
+
+1. Scans the `i18n` directory in the project and installed language packages for `.json` files that have a locale formatted file name
```

Comment:
We actually scan all `i18n` folders in modules that declare `i18n` support through the `specialFeatures` flag. We always scan the `venia-concept` directory (as it's always included). In a scaffolded project the `venia-concept` project becomes the scaffolded root directory.

Reply:
added step to the list about scanning the root project's directory